### PR TITLE
Fix TS name generation for names that start with `_`

### LINKED ISSUE #9447: Typescript enum naming regression
### What happened?

RDS resource in `pulumi-aws` is generating unexpected typescript constant names:
```diff
--- a/nodejs/types/enums/aws/rds/index.ts
+++ b/nodejs/types/enums/aws/rds/index.ts
@@ -17,11 +17,11 @@ export const AuroraServerlessEngine = {
 export type AuroraServerlessEngine = (typeof AuroraServerlessEngine)[keyof typeof AuroraServerlessEngine];
 
 export const MetricInterval = {
-    MetricInterval_15Second: 10,
-    MetricInterval_1Minute: 60,
-    MetricInterval_1Second: 1,
-    MetricInterval_30Second: 30,
-    MetricInterval_5Second: 5,
+    15Second_15Second: 10,
+    1Minute_1Minute: 60,
+    1Second_1Second: 1,
+    30Second_30Second: 30,
+    5Second_5Second: 5,
 } as const;
```

### Steps to reproduce

See above

### Expected Behavior

See above

### Actual Behavior

See above

### Versions used

v3.30.0

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by iwahbe**
# Description

Fixes #9447

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: pkg/codegen/testing/test/testdata/simple-enum-schema/python/pulumi_plant/_enums.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ class CloudAuditOptionsLogName(str, Enum):
     """
     What if triple quotes \"\"\" are used in the description
     """
+    CLOUD_AUDIT_OPTIONS_LOG_NAME_N_O_NAME = "_NO_NAME"
```
Comment (Frassle) [inline pkg/codegen/testing/test/testdata/simple-enum-schema/python/pulumi_plant/_enums.py:35]: N_O doesn't look right?

File: pkg/codegen/testing/test/testdata/simple-enum-schema/nodejs/types/enums/index.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export const CloudAuditOptionsLogName = {
      * What if triple quotes """ are used in the description
      */
     Synthetic: "SYNTHETIC",
+    CloudAuditOptionsLogName_NO_NAME: "_NO_NAME",
```
Comment (Frassle) [inline pkg/codegen/testing/test/testdata/simple-enum-schema/nodejs/types/enums/index.ts:28]: Why do we need the whole enum type name here? Identifiers are ok to start with underscore.

File: pkg/codegen/testing/test/testdata/simple-enum-schema/dotnet/Enums.cs [hunk 3]
```diff
@@ -36,6 +36,7 @@ private CloudAuditOptionsLogName(string value)
         /// What if triple quotes """ are used in the description
         /// </summary>
         public static CloudAuditOptionsLogName Synthetic { get; } = new CloudAuditOptionsLogName("SYNTHETIC");
+        public static CloudAuditOptionsLogName CloudAuditOptionsLogName_NO_NAME { get; } = new CloudAuditOptionsLogName("_NO_NAME");
```
Comment (Frassle) [inline pkg/codegen/testing/test/testdata/simple-enum-schema/dotnet/Enums.cs:39]: Why do we need the whole enum type name here? Identifiers are ok to start with underscore.

File: pkg/codegen/testing/test/testdata/simple-enum-schema/nodejs/types/enums/index.ts (hunk 2 above)
Comment (iwahbe) [reply pkg/codegen/testing/test/testdata/simple-enum-schema/nodejs/types/enums/index.ts:28]: I don't know. My suspicion is that it provides a way to guarantee uniqueness for when enums don't have unique namespaces. I don't know why this feature is present in nodejs. I'll make a note to clean this up for 4.0.

File: pkg/codegen/testing/test/testdata/simple-enum-schema/dotnet/Enums.cs (hunk 3 above)
Comment (iwahbe) [reply pkg/codegen/testing/test/testdata/simple-enum-schema/dotnet/Enums.cs:39]: Same as https://github.com/pulumi/pulumi/pull/9453#discussion_r857810830

> I don't know. My suspicion is that it provides a way to guarantee uniqueness for when enums don't have unique namespaces. I don't know why this feature is present in nodejs. I'll make a note to clean this up for 4.0.

File: pkg/codegen/testing/test/testdata/simple-enum-schema/python/pulumi_plant/_enums.py (hunk 1 above)
Comment (iwahbe) [reply pkg/codegen/testing/test/testdata/simple-enum-schema/python/pulumi_plant/_enums.py:35]: It doesn't look right, but it is orthogonal to this feature. I'd like to keep this PR focused. I'll open a separate issue to address this.

**Comment by AaronFriel** [comment]
Changelog will need a rebase, otherwise lgtm.

**Review by AaronFriel — APPROVED** [review]

**Review by t0yv0 — APPROVED** [review]

**Pull request merged**